### PR TITLE
introduce support for correlated vacuum

### PR DESCRIPTION
**Pull request by evaleev**
some orbitals will be neither fully occupied nor fully unoccupied. we assume that there is only a single instance of such space.

mbpt::mr::Operator will need to track creators/annihilators across at least 3 (active_occupied, active, and active_unoccupied) subspaces; if we need to relax inactive orbitals we will need to include inactive orbitals also

### CODE REVIEW

File: SeQuant/domain/mbpt/convention.cpp [hunk 1]
```diff
@@ -72,23 +72,49 @@ void register_standard_instances() {
...
+    IndexSpace::register_instance(declab(L"m"), IndexSpace::occupied, qnattr,
+                                  do_not_throw);
+    // introduced in MPQC LCAOWavefunction
     IndexSpace::register_instance(declab(L"e"), IndexSpace::unoccupied, qnattr,
                                   do_not_throw);
+    // introduced in MPQC for GF, CT-F12, and other ad hoc uses
     IndexSpace::register_instance(declab(L"x"), IndexSpace::all_active, qnattr,
                                   do_not_throw);
-    IndexSpace::register_instance(declab(L"p"), IndexSpace::all, qnattr,
+    // e.g. see DOI 10.1063/5.0067511
```
Comment (evaleev) [inline SeQuant/domain/mbpt/convention.cpp:108]: @Krzmbrzl do you know the origin of this convention?

Comment (Krzmbrzl) [reply SeQuant/domain/mbpt/convention.cpp:108]: As far as I know we have borrowed our notation mostly from internally-contracted CI theory but other than that, I don't know.
@ak-ustutt might know more though

Comment (ak-ustutt) [reply SeQuant/domain/mbpt/convention.cpp:108]: u, v, w, ....  or t, u, v, w, ... is rather common throughout MR works; u should be great here

File: SeQuant/domain/mbpt/convention.cpp [hunk 2]
```diff
@@ -72,23 +72,49 @@ void register_standard_instances() {
...
+    IndexSpace::register_instance(declab(L"u"), IndexSpace::active, qnattr,
                                   do_not_throw);
-    IndexSpace::register_instance(declab(L"α'"), IndexSpace::other_unoccupied,
-                                  qnattr, do_not_throw);
-    IndexSpace::register_instance(declab(L"α"), IndexSpace::complete_unoccupied,
-                                  qnattr, do_not_throw);
-    IndexSpace::register_instance(declab(L"κ"), IndexSpace::complete, qnattr,
+    // DOI 10.1063/5.0067511 uses I,J,K... and A,B,C... for these
```
[the remark this answers is not in the document]
Comment (Krzmbrzl) [reply SeQuant/domain/mbpt/convention.cpp:111]: @ak-ustutt told me that this is not really a standard per se. We are usually using it for our notation, but other than that Francesco Evangelista supposedly is (more or less) the only other person that consistently uses this notation.

File: SeQuant/domain/mbpt/mr.hpp [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+
+/// @param qns the quantum numbers to adjoint
+/// @return the adjoint of \p qns
+mbpt::mr::qns_t adjoint(mbpt::mr::qns_t);
+
+namespace mbpt {
+namespace mr {
+
+// clang-format off
+/// @brief makes a tensor-level fermionic many-body operator for use in single-reference methods
```
Comment (Krzmbrzl) [inline SeQuant/domain/mbpt/mr.hpp:110]:
```suggestion
/// @brief makes a tensor-level fermionic many-body operator for use in multireference methods
```

File: SeQuant/domain/mbpt/mr.hpp [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+
+/// A many-body operator has the following generic form:
+/// \f$ \frac{1}{P} T_{b_1 b_2 \dots b_B}^{k_1 k_2 \dots k_K} A^{b_1 b_2 \dots b_B}_{k_1 k_2 \dots k_K} \f$
+/// where \f$ \{B,K\} \f$ are number of bra/ket indices of \f$ T \f$ or, equivalently, the number of creators/annihilators
+/// of normal-ordered (w.r.t. the default, not necessarily Fermi vacuum) operator \f$ A \f$.
+/// Indices \f$ \{ b_i \} \f$ / \f$ \{ k_i \} \f$ are (active) unoccupied/occupied for (pure) excitation operators,
+/// are occupied/unoccupied for deexcitation operators; for general operators complete basis indices are assumed by default,
+/// unless overridden by user manually. \f$ P \f$ is the "normalization" factor and depends on the vacuum used to define \f$ A \f$,
```
Comment (Krzmbrzl) [inline SeQuant/domain/mbpt/mr.hpp:118]:
```suggestion
/// unless manually overridden by the user. \f$ P \f$ is the "normalization" factor and depends on the vacuum used to define \f$ A \f$,
```

File: SeQuant/domain/mbpt/mr.hpp [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+/// and indices \f$ \{ b_i \} \f$ / \f$ \{ k_i \} \f$.
+/// @note The choice of unoccupied indices/spaces can be controlled by the default Formalism:
+/// - if `get_default_formalism().sum_over_uocc() == SumOverUocc::Complete` IndexSpace::complete_unoccupied will be used instead of IndexSpace::active_unoccupied
+/// - if `get_default_formalism().csv() == CSVFormalism::CSV` will use cluster-specific (e.g., PNO) unoccupied indices
```
Comment (Krzmbrzl) [inline SeQuant/domain/mbpt/mr.hpp:122]: Does this apply to MR operators as well? I'm not sure that the concept can be straight-forwardly generalized to the MR case (though I also haven't looked into it).

**Pull request merged**